src/main/java/org/spongepowered/common/data/processor/data/entity/FlyingDataProcessor.java

```diff
@@ -0,0 +1,123 @@
+
+public class FlyingDataProcessor extends AbstractSpongeDataProcessor<FlyingData, ImmutableFlyingData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof net.minecraft.entity.Entity;
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    public Optional<FlyingData> from(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            final SpongeFlyingData flyingData = new SpongeFlyingData();
+            final boolean flying = ((net.minecraft.entity.Entity) dataHolder).isAirBorne;
+            return Optional.<FlyingData>of(flyingData.setFlying(flying));
+        } else {
+            return Optional.absent();
+        }
+
+    }
+
+    @Override
+    public Optional<FlyingData> fill(DataHolder dataHolder, FlyingData manipulator, MergeFunction overlap) {
+        if (supports(dataHolder)) {
+            final FlyingData merged = overlap.merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+            manipulator.set(Keys.IS_FLYING, merged.flying().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<FlyingData> fill(DataContainer container, FlyingData flyingData) {
+        flyingData.set(Keys.IS_FLYING, getData(container, Keys.IS_FLYING));
+        return Optional.of(flyingData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, FlyingData manipulator, MergeFunction function) {
+        // TODO
+        return null;
+    }
+
+    @Override
+    public Optional<ImmutableFlyingData> with(Key<? extends BaseValue<?>> key, Object value, ImmutableFlyingData immutable) {
+        // TODO
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            if (dataHolder instanceof EntityPlayer) {
+                EntityPlayer player = (EntityPlayer) dataHolder;
+                player.capabilities.allowFlying = false;
+            } else {
+                ((net.minecraft.entity.Entity) (dataHolder)).isAirBorne = false;
+            }
+            return DataTransactionBuilder.builder().result(DataTransactionResult.Type.SUCCESS).build();
+        }
+        return DataTransactionBuilder.builder().result(DataTransactionResult.Type.FAILURE).build();
+    }
+
+    @Override
+    public Optional<FlyingData> createFrom(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            boolean isFlying = ((net.minecraft.entity.Entity) dataHolder).isAirBorne;
+            if (dataHolder instanceof EntityPlayer) {
+                isFlying = ((EntityPlayer) dataHolder).capabilities.allowFlying;
+            }
+            return Optional.<FlyingData>of(new SpongeFlyingData(isFlying));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public int getPriority() {
```

Comment:
You're going to force me to add `final` to the `getPriority()` in `AbstractSpongeXProcessor`..... STOP OVERRIDING UNNECESSARILY!